File: sdks/cpp/examples/hello_world.cpp
/**
 * @file hello_world.cpp
 * @brief Hello World Example
 * 
 * The simplest possible Windjammer application in C++.
 * 
 * Build and run:
 *   mkdir build && cd build
 *   cmake ..
 *   cmake --build .
 *   ./examples/hello_world
 */

#include <windjammer/windjammer.hpp>
#include <iostream>

int main() {
    std::cout << "=== Windjammer Hello World (C++) ===\n";
    std::cout << "SDK Version: " << wj::VERSION << "\n\n";
    
    // Create a new application
    wj::App app;
    
    // Add a simple system
    app.add_system([]() {
        std::cout << "Hello from the game loop!\n";
    });
    
    std::cout << "Application created successfully!\n";
    std::cout << "Systems registered: 1\n\n";
    std::cout << "Note: Full app.run() would start the game loop\n";
    std::cout << "For this example, we're just demonstrating SDK setup\n\n";
    
    // Run the application
    app.run();
    
    return 0;
}

File: sdks/go/examples/hello_world.go
package main

import (
	"fmt"
	wj "github.com/windjammer/sdk-go/windjammer"
)

// Hello World Example
//
// The simplest possible Windjammer application in Go.
//
// Build and run:
//   go run examples/hello_world.go

func main() {
	fmt.Println("=== Windjammer Hello World (Go) ===")
	fmt.Println("SDK Version: 0.1.0\n")

	// Create a new application
	app := wj.NewApp()

	// Add a simple system
	app.AddSystem(func() {
		fmt.Println("Hello from the game loop!")
	})

	fmt.Println("Application created successfully!")
	fmt.Println("Systems registered: 1\n")
	fmt.Println("Note: Full app.Run() would start the game loop")
	fmt.Println("For this example, we're just demonstrating SDK setup\n")

	// Run the application
	app.Run()
}

